pkg/generate/app/sourcelookup.go

```diff
@@ -378,7 +381,7 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		Binary: repo.binary,
 	}
 
-	if repo.Info() != nil && repo.Info().Dockerfile != nil {
+	if (repo.ignoreRepository || repo.forceAddDockerfile) && repo.Info() != nil && repo.Info().Dockerfile != nil {
```

Comment:
> addDockerfileToBuildConfig :)

I started with that ;)
However, the problem is that that name lies. There are cases when we add the Dockerfile to the BC and this bool is false (and this is obvious by looking at the if condition).

I'll quote @smarterclayton here:

> Names are names - they get used a small fraction of the time.  Worrying
about names too much is just as bad as worrying not enough about names.

So let's not bikeshed on this. I'm willing to clean this up in a follow up.